legend-pure-functions/legend-pure-functions-relation/legend-pure-runtime-java-extension-interpreted-functions-relation/src/main/java/org/finos/legend/pure/runtime/java/extension/external/relation/interpreted/natives/shared/Shared.java

```diff
@@ -54,17 +60,67 @@ public TestTDS getTDS(CoreInstance value, ProcessorSupport processorSupport)
                 new TestTDSInterpreted(readCsv((value.getValueForMetaPropertyToOne("csv")).getName()), repository, processorSupport);
     }
 
-    public RelationType<?> getRelationType(ListIterable<? extends CoreInstance> params, int i)
+    public RelationType<?> getRelationType(ListIterable<? extends CoreInstance> params, int i, ProcessorSupport processorSupport)
```

Comment:
@pierredebelen this change is arguable a hack to make `toString()` works with `Relation<Any>`, you can see I added test which asserts on the resultant TDS after pivoting, the result is of type `Relation<Any>` and when calling `toString()` it will barf as underneath, it calls `->columns()`, and `columns()` break when calling on `Relation<Any>`.
Generally, I have tried the following example of casting and it just doesn't work
```
print(#TDS
    city, country, year, treePlanted
    NYC, USA, 2011, 5000
    NYC, USA, 2000, 5000
    SAN, USA, 2000, 2000
    SAN, USA, 2011, 100
    LND, UK, 2011, 3000
    SAN, USA, 2011, 2500
    NYC, USA, 2000, 10000
    NYC, USA, 2012, 7600
    NYC, USA, 2012, 7600
  #->pivot(~[country, city], ~[sum : x | $x.treePlanted : y | $y->plus(), count : x | $x : y | $y->size()])->cast(@Relation<(year: Integer)>)->toString());
```
This will result in error
```
Execution error at (resource:/welcome.pure line:50 column:124), "Cast exception: Relation<Any> cannot be cast to Relation<(year:Integer)>"
```
So I'm not sure how we can make `cast(@Relation<(...)>)` work with `Relation<Any>`. If I miss something, please let me know. Thanks!

Reply:
Reverted as we should not hack this way. Instead, we make sure `Pivot` in interpreted return the proper generic type at execution time.